protocol/kafka_sarama/v2/receiver.go

```diff
@@ -53,19 +53,25 @@ func (r *Receiver) Close(context.Context) error {
 }
 
 func (r *Receiver) ConsumeClaim(session sarama.ConsumerGroupSession, claim sarama.ConsumerGroupClaim) error {
-	for message := range claim.Messages() {
-		msg := message
-		m := NewMessageFromConsumerMessage(msg)
-
-		r.incoming <- msgErr{
-			msg: binding.WithFinish(m, func(err error) {
-				if protocol.IsACK(err) {
-					session.MarkMessage(msg, "")
-				}
-			}),
+	for {
+		select {
+		case msg, ok := <-claim.Messages():
+			if !ok {
+				return nil
+			}
+			m := NewMessageFromConsumerMessage(msg)
+
+			r.incoming <- msgErr{
+				msg: binding.WithFinish(m, func(err error) {
+					if protocol.IsACK(err) {
+						session.MarkMessage(msg, "")
+					}
+				}),
+			}
+		case <-session.Context().Done():
+			return nil
```

Comment:
nit: I generally recommend the context error here instead of surpressing it, e.g. `return session.Context().Err()`

Reply:
+1

Reply:
Added a comment in code for this, please review.

Reply:
You're the expert on the implementation so I think it's ok. Probably better to put that comment in the Method comment section so it's exported and doc-ed with the method?

Reply:
No, I am not an expert, just wanted that fix patched here. Let me move the comment.